api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,139 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	deployID := generateEntityID(req.DeployName)
+	familyName := addLayer0Prefix(d.Config.Instance(), req.DeployName)
+
+	taskDefinition, err := d.renderTaskDefinition(req.DeployFile, familyName)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefinitionOutput, err := d.createTaskDefinition(taskDefinition)
+	if err != nil {
+		return nil, err
+	}
+
+	bytes, err := json.Marshal(taskDefinitionOutput)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to extract deploy file: %s", err.Error())
+	}
+
+	deploy := &models.Deploy{
+		DeployName: req.DeployName,
+		DeployID:   deployID,
+		Version:    strconv.FormatInt(aws.Int64Value(taskDefinitionOutput.Revision), 10),
+		DeployFile: bytes,
+	}
+
+	taskDefinitionArn := aws.StringValue(taskDefinitionOutput.TaskDefinitionArn)
```

Comment:
[nit] go best-practice is to capitalize abbreviations: `taskDefinitionARN`

Reply:
I know we've been over this before, but it ends up being a weird translation to/from AWS, who use `Arn`